fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/world/poi/PointOfInterestHelper.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.object.builder.v1.world.poi;
+
+import java.util.function.Predicate;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.poi.PointOfInterestType;
+
+import net.fabricmc.fabric.mixin.object.builder.PointOfInterestTypeAccessor;
+
+/**
+ * This class provides utilities to create a {@link PointOfInterestType}.
+ */
+public final class PointOfInterestHelper {
```

Comment:
This class is rather odd, why does it mix create and register method names for no obvious reason?

A standalone `create` sounds like it'd create an instance of the containing class, `register` is similarly a bit too context-less for my taste.

The implementation looks a lot like it deserves a common (private?) method that does the actual registration with the other methods only restructuring the inputs.

Reply:
I could unbind the registration from this class, but the setup calls inside are necessary.

I'll look into overloading most of the method calls

Reply:
Everything is a register method now and called register. I've overloaded most methods down to two